src/energyplus/Test/ZoneMixing_GTest.cpp

```diff
@@ -0,0 +1,500 @@
+
+    auto zms = w.getObjectsByType(IddObjectType::ZoneMixing);
+    ASSERT_EQ(1, zms.size());
+    auto& zm = zms.front();
+
+    auto name = zm.nameString();
+    auto receiving_ = zm.getTarget(ZoneMixingFields::ZoneorSpaceName);
+    ASSERT_TRUE(receiving_);
+    auto source_ = zm.getTarget(ZoneMixingFields::SourceZoneorSpaceName);
+    ASSERT_TRUE(source_);
+    auto sch_ = zm.getTarget(ZoneMixingFields::ScheduleName);
+    ASSERT_TRUE(sch_);
+
+    EXPECT_EQ(receiving_->iddObject().type(), IddObjectType::Space);
+    EXPECT_EQ(source_->iddObject().type(), IddObjectType::Space);
+
+    EXPECT_EQ(name, zmAbs.nameString());
+    EXPECT_EQ(alwaysOnContinuousSchedule.nameString(), sch_->nameString());
+    EXPECT_EQ(space1.nameString(), receiving_->nameString());
+    EXPECT_EQ("Flow/Zone", zm.getString(ZoneMixingFields::DesignFlowRateCalculationMethod).get());
+    EXPECT_EQ(m3sAbs, zm.getDouble(ZoneMixingFields::DesignFlowRate).get());
+    EXPECT_TRUE(zm.isEmpty(ZoneMixingFields::FlowRateperFloorArea));
+    EXPECT_TRUE(zm.isEmpty(ZoneMixingFields::FlowRateperPerson));
+    EXPECT_TRUE(zm.isEmpty(ZoneMixingFields::AirChangesperHour));
+    EXPECT_EQ(space2.nameString(), source_->nameString());
+    EXPECT_TRUE(zm.isEmpty(ZoneMixingFields::DeltaTemperature));
+    EXPECT_TRUE(zm.isEmpty(ZoneMixingFields::DeltaTemperatureScheduleName));
+    EXPECT_TRUE(zm.isEmpty(ZoneMixingFields::MinimumReceivingTemperatureScheduleName));
+    EXPECT_TRUE(zm.isEmpty(ZoneMixingFields::MaximumReceivingTemperatureScheduleName));
+    EXPECT_TRUE(zm.isEmpty(ZoneMixingFields::MinimumSourceTemperatureScheduleName));
+    EXPECT_TRUE(zm.isEmpty(ZoneMixingFields::MaximumSourceTemperatureScheduleName));
+    EXPECT_TRUE(zm.isEmpty(ZoneMixingFields::MinimumOutdoorTemperatureScheduleName));
+    EXPECT_TRUE(zm.isEmpty(ZoneMixingFields::MaximumOutdoorTemperatureScheduleName));
+  }
+
+  {
+    ft.setExcludeSpaceTranslation(true);
+
+    Workspace w = ft.translateModel(m);
+
+    auto zones = w.getObjectsByType(IddObjectType::Zone);
+    ASSERT_EQ(1, zones.size());
+    auto zone = zones[0];
+    EXPECT_EQ(0, w.getObjectsByType(IddObjectType::Space).size());
+
+    auto zms = w.getObjectsByType(IddObjectType::ZoneMixing);
+    ASSERT_EQ(0, zms.size());
+
+    // This is problematic, we actually try to FT that object 4 TIMES
+    // EXPECT_EQ(1, ft.warnings().size());
+    EXPECT_GE(ft.warnings().size(), 1);
+    std::string expectedWarning =
+      "Object of type 'OS:ZoneMixing' and named 'zmAbs' doesn't have a Source Zone or Space, it will not be translated. As you were using "
+      "Space-Level ZoneMixing, and you are not translating to Spaces, it's possible it was pointing to two spaces inside the same zone";
+    auto warnings = ft.warnings();
+    EXPECT_NE(warnings.cend(),
+              std::find_if(warnings.cbegin(), warnings.cend(), [&expectedWarning](auto& l) { return l.logMessage() == expectedWarning; }));
+  }
+}
```

Comment:
Make a Space Level ZoneMixing that points to two spaces **inside the same zone**:
* When trnaslating to spaces: it should work fine
* When excluding spaces: it shouldn't translate it and we should get a nice warning.

Note that I noticed we try to translate it 4 times and opened https://github.com/NREL/OpenStudio/issues/4704